### DISCUSSION TITLE
Process limits during the creation of placeholder-variants for pre-existing photos

### DISCUSSION THREAD

**Discussion by Gendra13** [LycheeOrg/Lychee#2691] [Q&A] [answered]
I recently tested the current master version (0f32f23) and tried the new placeholder feature.

During my testing I came up with a question on how those placeholder images are generated for already existing photos:
The two ways that seem to exists are either via the _maintenance page_ or via the command line (`php artisan lychee:generatethumbs placeholder` and `php artisan lychee:encode_placeholders`). However, both ways only seem to process a limited number of photos per  run.

On the command line `php artisan lychee:generatethumbs placeholder` creates 100 placeholders per run and `php artisan lychee:encode_placeholders` only encodes 5 of them in each run. A similar limit seem to apply to the button on the maintenance page, I believe it is also limited to 100 images per “click”.

So for pre-existing galleries with thousands of photos you either have to _click-wait-click-wait-click…_ to create the placeholders for all images or fiddle around on the command line with a bash loop to run the commands hundreds of times, which is both quite cumbersome.

Is it really intended to be limited like this or am I missing a better way to handle the old images. If not, can those limits be raised or even removed?

**Comment by ildyria** [comment, marked as answer]
You can use the command line for a faster process:
```sh
docker exec -it Lychee-Core php artisan lychee:generate_thumbs placeholder 1000
```
or:
```sh
php artisan lychee:generate_thumbs placeholder 1000
```
Maybe we can consider to add a setting for those default 100 values.

if you feel like less worried about your process you can even go:
```
php artisan lychee:generate_thumbs placeholder 10000 10000
```

If you do `php artisan lychee:generate_thumbs` you will see the different options available.

**Reply by ildyria** [reply]
Unfortunately, if your process crash because of some obscure error, it is not possible to generate the following images...
As a result I had to do https://github.com/LycheeOrg/Lychee/pull/2690 which is waiting for review :)

**Reply by Gendra13** [reply]
Thank you. I didn't realize that you can specify the limit in timeout on the command line. This makes the migration much easier.
